cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -333,14 +333,13 @@ private WorkflowSpecification getWorkflowSpecForValidRun(String namespaceId, Str
     }
 
     WorkflowSpecification workflowSpec = appSpec.getWorkflows().get(workflowId);
-    ProgramId programId = new ProgramId(namespaceId, applicationId, ProgramType.WORKFLOW, workflowId);
+    ProgramId programId = new ProgramId(appId, ProgramType.WORKFLOW, workflowId);
     if (workflowSpec == null) {
       throw new ProgramNotFoundException(programId);
     }
 
     if (store.getRun(programId.run(runId)) == null) {
-      throw new NotFoundException(new ProgramRunId(programId.getNamespace(), programId.getApplication(),
-                                                   programId.getType(), programId.getProgram(), runId));
+      throw new NotFoundException(new ProgramRunId(appId, programId.getType(), programId.getProgram(), runId));
```

Comment:
Should store the value of `programId.run(runId)` before the `if` and use it in the `if` and here too.

Reply:
I'll take a look

Reply:
addressed by @jingtaowu3